client/src/app/components/users/users.component.ts

```diff
@@ -54,8 +58,35 @@ export class UsersComponent implements OnInit {
       }]
     } as User];
 
+    this.sortedData = this.users.slice();
+
    }
 
+   sortData(sort: Sort) {
+    const data = this.users.slice();
+    if (!sort.active || sort.direction == '') {
+      this.sortedData = data;
+      return;
+    }
+
+    this.sortedData = data.sort((a, b) => {
+      let isAsc = sort.direction == 'asc';
+      switch (sort.active) {
+        case 'id': return compare(a.id, b.id, isAsc);
+        case 'firstName': return compare(+a.firstName, +b.firstName, isAsc);
```

Comment:
Why did you add the `+` in front of `a.firstName`.  I *think* this is the javascript shorthand way of trying to convert a string to a number.